### ISSUE TITLE
eslint blocks legitimate errors in HTML context

### ISSUE THREAD

**Issue by 0sinner** [roadhump/SublimeLinter-eslint#81]
I have an .php file that is mostly HTML at this point. Inside I have tried `<script></script>` and `<script src="..."></script>`.

This resulted in an error along the lines of `Unexpected token '<'` from eslint (verified using sublimelinter debug mode).

So then I went into my project's root folder and tried to do `eslint --init`, answered all the questions and ended up with the following as my .eslintrc file.

```YAML
rules:
  indent:
    - 2
    - tab
  quotes:
    - 2
    - single
  linebreak-style:
    - 2
    - unix
  semi:
    - 2
    - always
env:
  browser: true
extends: 'eslint:recommended'
ecmaFeatures:
  jsx: true
  experimentalObjectRestSpread: true
plugins:
  - react
```

Opening the file again, eslint now gives me: `Error - missing semicolon. (semi)`

These errors are false positives, and block me being able to see errors from other linters. I don't plan to have much (if any) inline js code in my html/php files, so a good workaround for me would be to disable eslint in html contexts if that's possible. Of course I'd like it best if the solution made it work for all applicable contexts.

To me the error seems to be the selectors (html selector being the only one) set in `linter.py`. The selector (at least as far as I can tell) doesn't strip the `<script></script>` tags and eslint isn't expecting to start in an HTML context.

Please let me know if I can be of further assistance in troubleshooting/fixing this. Also it's entirely possible I'm just dumb and am missing some configuration flag.

EDIT: minor formatting edit and english, and clarity

**Comment by roadhump** [comment]
Hello

Did you check this doc http://www.sublimelinter.com/en/latest/usage.html?highlight=script#customized-syntax-definitions ? Probably there is a solution.

**Comment by 0sinner** [comment]
@roadhump thanks for the quick reply!

I looked through that document, but I'm missing it's relevance to my issue... Are you suggesting that I modify the HTML syntax definition files? I wonder if there is no better solution.. I can't imagine I'm the only one that has come across this issue.

Maybe I'm just being thick and missing something?

**Comment by roadhump** [comment]
This doc means that your HTML syntax file will be patched after SublimeLinter is installed and it will make linters work with code inside `<script>` tags. I suggest to restart ST if you are not yet. Also, is it possible that you use any custom, non-default HTML-syntax plugin?

**Comment by 0sinner** [comment]
Just restarted again, no dice.

I don't think I'm using anything special.. Maybe PHP syntax definitions? But I think those are built into ST3, I don't recall downloading anything like that.

Scanning through my loaded packages I found PHP CodeBeautifier in the list. I disabled it, restarted again, and still getting the same errors.

Is there some way I can check which HTML and HTML (Rails) syntax definitions are loaded?

**Comment by 0sinner** [comment]
Oh wait. Maybe I need to make whatever fix was done to the HTML and HTML (Rails) languages to the PHP language definition?

**Comment by 0sinner** [comment]
Looking at PHP.tmlanguage it looks like it relies on the underlying HTML language definitions instead. I'm not seeing any reference to `<script` or `&lt;script` So I'm not seeing the issue.

As a test I also renamed the file to be index.html, reloaded ST3, and I'm still getting the same issue.

**Comment by roadhump** [comment]
It is strange, I've tested html and php syntax and it seems work. Probably reinstall SublimeLinter will help.

**Comment by 0sinner** [comment]
Just reinstalled..

For now I'm using PackageResourceViewer, opening SublimeLinter-contrib-eslint\linter.py and removing 'html' from the list to stop it from linting using eslint in this context. But I am interested in continuing to troubleshoot.

I recently (within the past week) refreshed my PC (completely clean reinstall of windows). At that point I installed the latest version of sublime and started installing linters (I had never used linters in sublime prior to this).

Maybe fresh installs of sublime, sublimelinter or sublimelinter-contrib-eslint, cause some kind of issue? Maybe I just screwed up the configuration in some way?

I'd be happy to continue troubleshooting this, if you need me to do some debugging on my end to determine what is different between your environment and mine.

**Comment by roadhump** [comment]
Try another SublimeLinter js plugins (jshint or jscs). If they work the same way in HTML files, I think you should post this issue to SublimeLinter itself.
